### PR TITLE
added fixed samples to paramter sweep reader and looptool

### PR DESCRIPTION
**Pull request by avdudchenko**
## Fixes/Resolves:

Adds predetermined fixed and random samples to parameter sweep reader and support in loop tool

(replace this with the issue # fixed or resolved, if no issue exists then a brief statement of what this PR does)

## Changes proposed in this PR:
-
-

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Review by k1nshuk — APPROVED** [review]
LGTM

File: watertap/tools/analysis_tools/loop_tool/loop_tool.py [hunk 1]
```diff
@@ -402,9 +402,18 @@ def get_diff_params(self, key, loop_value):
             force_rerun,
         )
 
+    def get_num_samples(self, value_dict):
+        num_samples = value_dict.get("num_samples")
+        if value_dict.get("num_samples") is not None:
+            return value_dict.get("num_samples")
+        elif value_dict.get("array") is not None:
+            return len(value_dict.get("array"))
```
Comment (bknueven) [inline watertap/tools/analysis_tools/loop_tool/loop_tool.py]: This isn't tested -- which surprises me given the test below.

File: watertap/tools/analysis_tools/loop_tool/loop_tool.py [hunk 2]
```diff
@@ -402,9 +402,18 @@ def get_diff_params(self, key, loop_value):
...
+        else:
+            raise "Provide a valid sweep type, refer to parameter_sweep_reader for valid types"
```
Comment (bknueven) [inline watertap/tools/analysis_tools/loop_tool/loop_tool.py]:
```suggestion
        num_samples = value_dict.get("num_samples")
        array = value_dict.get("array")
        if num_samples is not None and array is not None:
            raise RuntimeError("Supply `num_samples` or `array`, not both")
        if num_samples is not None:
            return num_samples
        if array is not None:
            return len(array)
        raise RuntimeError("Provide a valid sweep type, refer to parameter_sweep_reader for valid types")
```
I've made this method catch another edge case to provide a more robust implementation.

File: watertap/tools/analysis_tools/loop_tool/loop_tool.py (hunk 1 above)
Comment (avdudchenko) [reply watertap/tools/analysis_tools/loop_tool/loop_tool.py]: I don't think I have a test configured to run tests on predifined fixed or random sample in loop tool.  I guess I should add a test fo rthis.  EDIT: Added test for this.

**Review by bknueven — APPROVED** [review]

**Pull request merged**